### ISSUE TITLE
TV, EPG data

### ISSUE THREAD

**Issue by redjames** [calabash/calabash-android#895]
Hi, everyone. This is not an issue, but I need some help with an element. I have to test a tv element in my app and acces to the epg to validate the channels. Is there a way to test this type of elements?
![image](https://user-images.githubusercontent.com/17856696/46301157-29e75880-c56b-11e8-8b0f-02a8e9a844cf.png)

Thanks in advance

**Comment by sherbhachu** [comment]
looks like you have the necessary details, package, class and id.

**Comment by redjames** [comment]
Yeah, but the thats the class, id, etc for the whole element. What I need is to interact with the channels, schedules. Navigate through them. When I got the app, I was thinking that one channels and contents will be and elements. This is an example of an android epg:

![image](https://user-images.githubusercontent.com/17856696/46301786-fc9baa00-c56c-11e8-872a-a1a5cccc499e.png)

**Comment by sherbhachu** [comment]
Ah, ok, I see what you mean.

You may need to combine Calabash with an OCR library, that’s what I have
done in the past.

**Comment by redjames** [comment]
Thanks for the reply. 

Do you have an example?

**Comment by sherbhachu** [comment]
Unfortunately not, it’s a fairly significant piece of work.

What I did was use the ocr tool t capture the screen and then extract the
words. You then use that to extract and create a map of the words on the
screen along with their x and y coordinates. You then use calabash to
undertake an action on that coordinate.

Another option is of course to write your own ‘driver’.

Yet another option is to use a ruby implementation of Sikuli, something
like this...


https://codoid.com/using-sikuli-from-ruby/
